bundles/org.openhab.binding.carnet/src/main/java/org/openhab/carnet/internal/services/CarNetVehicleServiceDestinations.java

```diff
@@ -0,0 +1,133 @@
+import org.openhab.binding.carnet.internal.api.CarNetApi;
+import org.openhab.binding.carnet.internal.api.CarNetApiGSonDTO.CNDestinations.CarNetDestination;
+import org.openhab.binding.carnet.internal.api.CarNetApiGSonDTO.CNDestinations.CarNetDestinationList;
+import org.openhab.binding.carnet.internal.handler.CarNetVehicleHandler;
+import org.openhab.binding.carnet.internal.provider.CarNetIChanneldMapper.ChannelIdMapEntry;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * {@link CarNetVehicleServiceDestinations} implements the destination hostory
+ *
+ * @author Markus Michels - Initial contribution
+ */
+public class CarNetVehicleServiceDestinations extends CarNetVehicleBaseService {
+    public CarNetVehicleServiceDestinations(CarNetVehicleHandler thingHandler, CarNetApi api) {
+        super(thingHandler, api);
+        serviceId = CNAPI_SERVICE_MY_AUDI_DESTINATIONS;
+    }
+
+    @Override
+    public boolean createChannels(Map<String, ChannelIdMapEntry> channels) throws CarNetException {
+        if (getConfig().vehicle.numDestinations == 0) {
+            return false;
+        }
+        try {
+            update(channels);
+        } catch (CarNetException e) {
+
+        }
+        return true;
+    }
+
+    private boolean createChannels(Map<String, ChannelIdMapEntry> ch, int index) {
+        boolean a = false;
+        String group = CHANNEL_GROUP_DEST_PRE + index;
+        a |= addChannel(ch, group, CHANNEL_DEST_NAME, ITEMT_STRING, null, false, true);
+        a |= addChannel(ch, group, CHANNEL_DEST_POI, ITEMT_STRING, null, false, true);
+        a |= addChannel(ch, group, CHANNEL_DEST_GEO, ITEMT_LOCATION, null, false, true);
+        a |= addChannel(ch, group, CHANNEL_DEST_STREET, ITEMT_STRING, null, false, true);
+        a |= addChannel(ch, group, CHANNEL_DEST_CITY, ITEMT_STRING, null, false, true);
+        a |= addChannel(ch, group, CHANNEL_DEST_ZIP, ITEMT_STRING, null, false, true);
+        a |= addChannel(ch, group, CHANNEL_DEST_COUNTY, ITEMT_STRING, null, true, true);
+        a |= addChannel(ch, group, CHANNEL_DEST_SOURCE, ITEMT_STRING, null, true, true);
+        return a;
+    }
+
+    @Override
+    public boolean serviceUpdate() throws CarNetException {
+        return update(null);
+    }
+
+    private boolean update(@Nullable Map<String, ChannelIdMapEntry> channels) throws CarNetException {
+        boolean updated = false;
+        CarNetDestinationList dest = api.getDestinations();
+        if (dest.destination.size() == 0) {
+            // no/empty list
+            return false;
+        }
+
+        Collections.sort(dest.destination, Collections.reverseOrder(new Comparator<CarNetDestination>() {
+            @Override
+            public int compare(CarNetDestination a, CarNetDestination b) {
+                return a.id.compareTo(b.id);
+            }
+        }));
```

Comment:
You could `CarNetDestination` let implement `Comparable`. That'd be more elegant than using an anonymous class here.

Reply:
That goes beyond my Java skills, need advise

Reply:
Add `implements Comparable<CarNetDestination>` to CarNetDestination. Then, you can remove the anonymous inner Comparator class.